File: src/main/java/io/vertigo/shiny/components/card/ShinyCardComponent.java
package io.vertigo.shiny.components.card;

import io.vertigo.core.lang.Assertion;
import io.vertigo.shiny.components.ShinyComponent;

public record ShinyCardComponent(
		String title,
		String subtitle,
		String description,
		String imageUrl,
		String imageAlt,
		String link,
		String icon,
		String badgeLabel,
		String badgeColor,
		ShinyCardFormat format) implements ShinyComponent {

	public ShinyCardComponent {
		Assertion.check()
				.isNotBlank(title, "Card title cannot be blank");
	}
}

File: banshee/src/models/ShinyCardComponent.ts
import { ShinyComponent } from './ShinyComponent';
import { ShinyCardFormat } from './ShinyCardFormat';

export class ShinyCardComponent implements ShinyComponent {
  title: string;
  subtitle?: string;
  description?: string;
  imageUrl?: string;
  imageAlt?: string;
  link?: string;
  icon?: string;
  badgeLabel?: string;
  badgeColor?: string;
  format: ShinyCardFormat;
  type: string = 'card';

  constructor(
    title: string,
    subtitle: string,
    description: string,
    imageUrl: string,
    imageAlt: string,
    link: string,
    icon: string,
    badgeLabel: string,
    badgeColor: string,
    format: ShinyCardFormat
  ) {
    this.title = title;
    this.subtitle = subtitle;
    this.description = description;
    this.imageUrl = imageUrl;
    this.imageAlt = imageAlt;
    this.link = link;
    this.icon = icon;
    this.badgeLabel = badgeLabel;
    this.badgeColor = badgeColor;
    this.format = format;
  }
}
